### DISCUSSION TITLE
vcpkg builds package repeatedly

### DISCUSSION THREAD

**Discussion by 0O0o0oOoO00** [microsoft/vcpkg#41059] [Q&A]
Vcpkg always rebuild some packages once I changed dependencies, even if I specified binary cache dir in cmake and vcpkg has cached binary from last build
Here is my vcpkg.json

    {
        "name": "project",
        "version": "0.1.0",
        "dependencies": [
            "qt5-base",
            "detours",
            "boost-locale"
        ]
    }

And here is part of my CMakePresets.json

    "configurePresets": [
        {
            "name": "Default",
            "hidden": true,
            "environment": {
                "VCPKG_ROOT": "${sourceDir}/vcpkg",
                "VCPKG_DOWNLOADS": "${sourceDir}/vcpkg_packages/downloads",
                "VCPKG_DEFAULT_BINARY_CACHE": "${sourceDir}/vcpkg_packages/binary_cache",
                "VCPKG_BINARY_SOURCES": "files,${sourceDir}/vcpkg_packages/binary_cache",
                "VCPKG_PACKAGES_DIR": "${sourceDir}/vcpkg_packages"
            },
            "cacheVariables": {
                "CMAKE_AUTOUIC": "ON",
                "CMAKE_AUTORCC": "ON",
                "CMAKE_AUTOMOC": "ON",
                "CMAKE_TOOLCHAIN_FILE": "$env{VCPKG_ROOT}/scripts/buildsystems/vcpkg.cmake",
                "VCPKG_INSTALLED_DIR": "$env{VCPKG_PACKAGES_DIR}",
                "VCPKG_OVERLAY_TRIPLETS": "${sourceDir}/cmake",
                "VCPKG_INSTALL_OPTIONS": "--x-buildtrees-root=$env{VCPKG_PACKAGES_DIR}/buildtrees;--x-packages-root=$env{VCPKG_PACKAGES_DIR}/packages;--clean-buildtrees-after-build;--clean-packages-after-build",
                "CMAKE_MODULE_PATH": "${sourceDir}/cmake"
            }
        }
    ]

I have set `VCPKG_DEFAULT_BINARY_CACHE` and `VCPKG_BINARY_SOURCES`, but it seems doesn't work very well
Here is part of cmake output(snippets for brevity)

    -- Running vcpkg install
    Detecting compiler hash for triplet x64-windows...
    -- Automatically setting %HTTP(S)_PROXY% environment variables to "127.0.0.1:7890".
    Compiler found: D:/VisualStudio2022/VC/Tools/MSVC/14.41.34120/bin/Hostx64/x64/cl.exe
    The following packages will be rebuilt:
    * brotli:x64-windows@1.1.0#1
    <...snip...>
    * zstd:x64-windows@1.5.6
      The following packages will be built and installed:
    * boost-algorithm:x64-windows@1.85.0#1
    <...snip...>
    * vcpkg-boost:x64-windows@2024-05-15
      Additional packages (*) will be modified to complete this operation.
      Restored 0 package(s) from E:\ClionProjects\project\vcpkg_packages\binary_cache in 2.67 ms. Use --debug to see more details.
      Restored 0 package(s) from E:/ClionProjects/project/vcpkg_packages/binary_cache in 1.93 ms. Use --debug to see more details.
      Removing 1/96 qt5-base:x64-windows
      Elapsed time to handle qt5-base:x64-windows: 903 ms
    <...snip...>
      Removing 21/96 detours:x64-windows
      Elapsed time to handle detours:x64-windows: 5.5 ms
    <...snip(a lot of rebuild and install)...>
    -- Running vcpkg install - done
    -- Configuring done (1470.1s)
    -- Generating done (0.2s)
    -- Build files have been written to: E:/ClionProjects/project/cmake_build/Debug

**Comment by autoantwort** [comment]
It can happen that when you change dependencies different feature sets of the other dependencies gets resolved so that the other dependencies must be rebuild.

**Comment by ras0219-msft** [comment]
Please take a look at the [Binary Caching Troubleshooting Guide](https://learn.microsoft.com/vcpkg/users/binarycaching-troubleshooting) and let us know if you have any questions!

**Comment by 0O0o0oOoO00** [comment]
Oh, sorry, i forgot to close this. I solved this by reading [Binary Caching Troubleshooting Guide](https://learn.microsoft.com/vcpkg/users/binarycaching-troubleshooting). I forgot to give `readwrite` permission to binary cache setting
